clients/client/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.projectnessie.model.Contents.Type;
+
+/**
+ * This test merely checks the JSON serialization/deserialization of the model classes, with an
+ * intention to identify breaking cases whenever jackson version varies.
+ */
+public class TestModelObjectsSerialization {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @ParameterizedTest
+  @MethodSource("goodCases")
+  void testGoodSerDeCases(Case goodCase) throws JsonProcessingException {
+    String json = MAPPER.writeValueAsString(goodCase.obj);
+    Assertions.assertThat(json).isEqualTo(goodCase.deserializedJson);
+    Object deserialized = MAPPER.readValue(json, goodCase.deserializeAs);
+    Assertions.assertThat(deserialized).isEqualTo(goodCase.obj);
+  }
+
+  @ParameterizedTest
+  @MethodSource("negativeCases")
+  void testNegativeSerDeCases(Case invalidCase) {
+    assertThrows(
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(invalidCase.deserializedJson, invalidCase.deserializeAs));
+  }
+
+  static List<Case> goodCases() {
+    final Instant now = Instant.now();
+    final String hash = "3e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10e";
+    final String branchName = "testBranch";
+
+    return Arrays.asList(
+        new Case(
+            ImmutableBranch.builder().hash(hash).name(branchName).build(),
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\","
+                + "\"hash\":\"3e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10e\"}"),
```

Comment:
Should `Branch.class` be `Reference.class`, perhaps, to validate the handling of the `type` attribute?

Reply:
And add cases for the other `Reference` sub-types?

Reply:
Did you mean to add test cases for other `Reference` sub-types? I could not find any in the latest commit :thinking: